### PR TITLE
 FeedbackSessionAttributes class instance variables should have 'private' visibility #5433 

### LINKED ISSUE #5433: FeedbackSessionAttributes class instance variables should have 'private' visibility
Currently all of them are set to 'public'.

### PR DESCRIPTION
**Pull request by rclakmal**
Fixes #5433

### CODE REVIEW

**Comment by rclakmal** [comment]
This refactoring turned out to be little big :-)

**Comment by wkurniawan07** [comment]
@rclakmal the changes look good, but there's a merge conflict that needs resolve. Once you've fixed it tag me again and I'll have another look!

**Comment by rclakmal** [comment]
@wkurniawan07 Done

**Comment by wkurniawan07** [comment]
@rclakmal as suggested in #5477, some of the setters might not be needed. Try to identify which ones are not and remove them, if they are only set during construction then the constructor can contain the entire setting method.

**Comment by rclakmal** [comment]
@wkurniawan07 Done :)

File: src/client/java/teammates/client/scripts/DataMigrationForResponseRate.java [hunk 1]
```diff
@@ -66,7 +66,7 @@ private void updateRespondantsForAllSessions() {
         List<FeedbackSessionAttributes> feedbackSessionsWithNoRespondants = new ArrayList<FeedbackSessionAttributes>();
         
         for (FeedbackSessionAttributes feedbackSession : feedbackSessions) {
-            if (feedbackSession.respondingStudentList.size() != 0 || feedbackSession.respondingInstructorList.size() != 0) {
+            if (feedbackSession.getRespondingStudentList().size() != 0 || feedbackSession.getRespondingInstructorList().size() != 0) {
```
Comment (wkurniawan07) [inline src/client/java/teammates/client/scripts/DataMigrationForResponseRate.java]: Let's add a line break here, before the `||` operator.

File: src/client/java/teammates/client/scripts/OfflineBackup.java [hunk 2]
```diff
@@ -404,7 +404,7 @@ protected void saveFeedbackResponseComment(FeedbackResponseCommentAttributes fee
     }
     
     protected void saveFeedbackSession(FeedbackSessionAttributes feedbackSession) {
-        appendToFile(currentFileName, formatJsonString(feedbackSession.getJsonString(), feedbackSession.feedbackSessionName + "%" + feedbackSession.courseId));
+        appendToFile(currentFileName, formatJsonString(feedbackSession.getJsonString(), feedbackSession.getFeedbackSessionName() + "%" + feedbackSession.getCourseId()));
     }
```
Comment (wkurniawan07) [inline src/client/java/teammates/client/scripts/OfflineBackup.java]: Here as well.

File: src/main/java/teammates/logic/core/FeedbackSessionsLogic.java [hunk 3]
```diff
@@ -1679,20 +1679,20 @@ public FeedbackSessionDetailsBundle getFeedbackSessionDetails(
...
+                    fqLogic.getFeedbackQuestionsForInstructor(fsa.getFeedbackSessionName(), fsa.getCourseId(), fsa.getCreatorEmail());
             List<FeedbackQuestionAttributes> validQuestions =
-                    fqLogic.getQuestionsWithRecipients(instuctorQuestions, fsa.creatorEmail);
+                    fqLogic.getQuestionsWithRecipients(instuctorQuestions, fsa.getCreatorEmail());
             if (validQuestions.isEmpty()) {
                 break;
             }
             details.stats.expectedTotal = 1;
-            if (this.isFeedbackSessionFullyCompletedByInstructor(fsa.feedbackSessionName, fsa.courseId, fsa.creatorEmail)) {
+            if (this.isFeedbackSessionFullyCompletedByInstructor(fsa.getFeedbackSessionName(), fsa.getCourseId(), fsa.getCreatorEmail())) {
```
Comment (wkurniawan07) [inline src/main/java/teammates/logic/core/FeedbackSessionsLogic.java]: Might be good if you add a line break here, and if possible, remove the `this`.

File: src/main/java/teammates/logic/core/FeedbackSessionsLogic.java [hunk 4]
```diff
@@ -1679,20 +1679,20 @@ public FeedbackSessionDetailsBundle getFeedbackSessionDetails(
...
             
-            details.stats.submittedTotal += fsa.respondingStudentList.size() + fsa.respondingInstructorList.size();
+            details.stats.submittedTotal += fsa.getRespondingStudentList().size() + fsa.getRespondingInstructorList().size();
 
             break;
 
         case PRIVATE:
             List<FeedbackQuestionAttributes> instuctorQuestions =
-                    fqLogic.getFeedbackQuestionsForInstructor(fsa.feedbackSessionName, fsa.courseId, fsa.creatorEmail);
+                    fqLogic.getFeedbackQuestionsForInstructor(fsa.getFeedbackSessionName(), fsa.getCourseId(), fsa.getCreatorEmail());
```
Comment (wkurniawan07) [inline src/main/java/teammates/logic/core/FeedbackSessionsLogic.java]: A line break here would be good.

File: src/main/java/teammates/storage/api/FeedbackSessionsDb.java [hunk 5]
```diff
@@ -88,11 +88,11 @@ public void createFeedbackSessions(Collection<FeedbackSessionAttributes> feedbac
...
             
-            Date standardStart = TimeHelper.convertToUserTimeZone(startCal, fs.timeZone - zone).getTime();
-            Date standardEnd = TimeHelper.convertToUserTimeZone(endCal, fs.timeZone - zone).getTime();
+            Date standardStart = TimeHelper.convertToUserTimeZone(startCal, fs.getTimeZone() - zone).getTime();
+            Date standardEnd = TimeHelper.convertToUserTimeZone(endCal, fs.getTimeZone() - zone).getTime();
             
-            boolean isStartTimeWithinRange = TimeHelper.isTimeWithinPeriod(standardStart, standardEnd, fs.startTime, true, false);
-            boolean isEndTimeWithinRange = TimeHelper.isTimeWithinPeriod(standardStart, standardEnd, fs.endTime, false, true);
+            boolean isStartTimeWithinRange = TimeHelper.isTimeWithinPeriod(standardStart, standardEnd, fs.getStartTime(), true, false);
+            boolean isEndTimeWithinRange = TimeHelper.isTimeWithinPeriod(standardStart, standardEnd, fs.getEndTime(), false, true);
```
Comment (wkurniawan07) [inline src/main/java/teammates/storage/api/FeedbackSessionsDb.java]: Here as well.

File: src/main/java/teammates/storage/api/FeedbackSessionsDb.java [hunk 6]
```diff
@@ -553,6 +553,6 @@ private FeedbackSession getFeedbackSessionEntity(String feedbackSessionName, Str
     @Override
     protected Object getEntity(EntityAttributes attributes) {
         FeedbackSessionAttributes feedbackSessionToGet = (FeedbackSessionAttributes) attributes;
-        return getFeedbackSessionEntity(feedbackSessionToGet.feedbackSessionName, feedbackSessionToGet.courseId);
+        return getFeedbackSessionEntity(feedbackSessionToGet.getFeedbackSessionName(), feedbackSessionToGet.getCourseId());
```
Comment (wkurniawan07) [inline src/main/java/teammates/storage/api/FeedbackSessionsDb.java]: Here, too.

**Comment by wkurniawan07** [comment]
Hi @rclakmal, in general the changes are fine (naturally because the changes are of a simple nature), so it'll be great if you can identify places where it'd be good to add line breaks to prevent reviewers (this applies to all your other PRs as well) from needing to scroll right in the GitHub tabs :)

**Comment by rclakmal** [comment]
@wkurniawan07 Done :)

**Comment by wkurniawan07** [comment]
@rclakmal good job :+1: just a merge conflict to be resolved.

**Comment by rclakmal** [comment]
Fixed merge conflicts.

**Comment by wkurniawan07** [comment]
@rclakmal good job, I will be merging this immediately.

**Pull request merged**